checkov/common/runners/runner_registry.py

```diff
@@ -99,3 +114,74 @@ def enrich_report_with_guidelines(scan_report, guidelines):
             if record.check_id in guidelines:
                 record.set_guideline(guidelines[record.check_id])
+
+    @staticmethod
+    def get_enriched_resources(report, repo_root):
+        parser = Parser()
+        tf_definitions = {}
+        parsing_errors = {}
+        Parser.parse_directory(
+            parser,
+            directory=repo_root,  # assume plan file is in the repo-root
+            out_definitions=tf_definitions,
+            out_parsing_errors=parsing_errors,
+        )
+
+        enriched_resources = defaultdict(dict)
+        for definition in tf_definitions.items():
+            definitions_context = parser_registry.enrich_definitions_context(definition)
+
+        for full_file_path, definition in tf_definitions.items():
+            abs_scanned_file, _ = tf_runner._strip_module_referrer(full_file_path)
+            scanned_file = os.path.relpath(abs_scanned_file, repo_root)
+            for block_type in definition.keys():
+                if block_type in CHECK_BLOCK_TYPES:
+                    for entity in definition[block_type]:
+                        context_parser = parser_registry.context_parsers[block_type]
+                        definition_path = context_parser.get_entity_context_path(entity)
+                        entity_id = ".".join(definition_path)
+                        entity_context_path = [block_type] + definition_path
+                        entity_context = dict_utils.getInnerDict(
+                            definitions_context[full_file_path], entity_context_path
+                        )
+                        entity_lines_range = [
+                            entity_context.get("start_line"),
+                            entity_context.get("end_line"),
+                        ]
+                        entity_code_lines = entity_context.get("code_lines")
+                        skipped_checks = entity_context.get("skipped_checks")
+                        enriched_resources[entity_id] = {
+                            "entity_code_lines": entity_code_lines,
+                            "entity_lines_range": entity_lines_range,
+                            "scanned_file": scanned_file,
+                            "skipped_checks": skipped_checks
+                        }
+        return enriched_resources
+
+    @staticmethod
+    def enrich_plan_report(report, enriched_resources):
+        # This enriches reports with the appropriate filepath, line numbers, and codeblock
+        for record in report.failed_checks:
+            if record.resource in enriched_resources:
+                record.file_path = enriched_resources[record.resource]["scanned_file"]
+                record.file_line_range = enriched_resources[record.resource][
+                    "entity_lines_range"
+                ]
+                record.code_block = enriched_resources[record.resource][
+                    "entity_code_lines"
+                ]
```

Comment:
instead of retrieving several times the same value from the dict you can change it to retrieve it once
```python
        for record in report.failed_checks:
            enriched_resource = enriched_resources.get(record.resource)
            if enriched_resource:
                record.file_path = enriched_resource["scanned_file"]
                record.file_line_range = enriched_resource["entity_lines_range"]
                record.code_block = enriched_resource[entity_code_lines"]
```
when this would be a Python 3.8+ project, then you could even use the walrus operator `:=` 😄

Reply:
:) will take time to support python 3.8 only. 
We saw some envs with python 3.6 or lower that needed checkov.

Reply:
let's go directly up to 3.10 only 😄 
@schosterbarak I hope it is ok, I did the code review here?

Reply:
always good to have another set of eyes. Maybe we can migrate some of the comments to a github suggestion: http://haacked.com/archive/2019/06/03/suggested-changes/ ? 
It might make the changes slightly easier.

Reply:
you are right, some of them can be made as a suggestion, didn't think about using suggestions 😄